test/secret-storage/SecretManager.ts

```diff
@@ -0,0 +1,1427 @@
+
+        await stakingToken.transfer(addrs[1], 10n ** 20n);
+        await stakingToken.connect(signers[1]).approve(secretStore.target, 10n ** 20n);
+
+        // REGISTER SECRET STORE ENCLAVES
+        const timestamp = await time.latest() * 1000;
+        let signTimestamp = await time.latest();
+        let storageCapacity = 1e9,
+            stakeAmount = parseUnits("10");	// 10 POND
+        for (let index = 0; index < 3; index++) {
+            let [attestationSign, attestation] = await createAttestation(
+                pubkeys[17 + index],
+                image2,
+                wallets[14],
+                timestamp - 540000
+            );
+
+            let signedDigest = await createSecretStoreSignature(addrs[1], storageCapacity, signTimestamp,
+                wallets[17 + index]);
+
+            await secretStore.connect(signers[1]).registerSecretStore(
+                attestationSign,
+                attestation,
+                storageCapacity,
+                signTimestamp,
+                signedDigest,
+                stakeAmount
+            );
+        }
+
+        // CREATE SECRET
+        let sizeLimit = 1000,
+            endTimestamp = await time.latest() + 800,
+            usdcDeposit = parseUnits("30", 6);
+        await secretManager.createSecret(sizeLimit, endTimestamp, usdcDeposit);
+
+        let secretId = 1,
+            signedDigest = await createAcknowledgeSignature(secretId, signTimestamp, wallets[17]);
+        await secretManager.acknowledgeStore(secretId, signTimestamp, signedDigest);
+    });
+
+    takeSnapshotBeforeAndAfterEveryTest(async () => { });
+
+    it("can terminate the secret before end timestamp", async function () {
+        await time.increase(100);
+        let secretId = 1;
+        let usdcRefund = 700 * 1000 * 10 * 3; // reductionInDuration * sizeLimit * feeRate * noOfNodes
+        let secretManagerInitialBal = await usdcToken.balanceOf(secretManager.target);
+        let initialUsdcDeposit = (await secretManager.userStorage(secretId)).usdcDeposit;
+
+        await expect(secretManager.terminateSecret(secretId))
+            .to.emit(secretManager, "SecretTerminated").withArgs(secretId);
+        expect((await secretManager.userStorage(secretId)).endTimestamp).to.eq(await time.latest());
+
+        let secretManagerFinalBal = await usdcToken.balanceOf(secretManager.target);
+        let finalUsdcDeposit = (await secretManager.userStorage(secretId)).usdcDeposit;
+        expect(secretManagerInitialBal - secretManagerFinalBal).to.eq(usdcRefund);
+        expect(initialUsdcDeposit - finalUsdcDeposit).to.eq(usdcRefund);
+    });
+
+    it("cannot update end timestamp without secret owner", async function () {
+        let secretId = 1;
+        await expect(secretManager.connect(signers[1]).terminateSecret(secretId))
+            .to.be.revertedWithCustomError(secretManager, "SecretManagerNotUserStoreOwner");
+    });
```

Comment:
few more cases:
1. termination after termination
2. termination after end timestamp
3. combination of termination and acknowledgement time period